libs/src/services/ethContracts/index.js

```diff
@@ -158,11 +181,12 @@ class EthContracts {
    * @param {string} spType service provider type: 'discovery-provider' | 'content-service' | 'creator-node'
    * @return {Promise<string>} A valid service provider url with the fastest response
    */
-  async selectLatestServiceProvider (spType, whitelist = null) {
+  async selectLatestVersionServiceProvider (spType, whitelist = null) {
     let serviceProviders = await this.ServiceProviderFactoryClient.getServiceProviderList(spType)
     if (whitelist) {
       serviceProviders = serviceProviders.filter(d => whitelist.has(d.endpoint))
     }
+    console.info('Looking latest for service provider in', serviceProviders)
```

Comment:
lets move this to after the initialization of 'expectedVersion' and include that in the log statement